lib/Service/ApiService.php

```diff
@@ -136,25 +136,19 @@ public function create(?int $fileId = null, ?string $filePath = null, ?string $t
 
 		$session = $this->sessionService->initSession($document->getId(), $guestName);
 
+		$documentState = null;
+		$content = null;
 		if ($freshSession) {
 			$this->logger->debug('Starting a fresh editing session for ' . $file->getId());
-			$documentState = null;
 			$content = $this->loadContent($file);
 		} else {
 			$this->logger->debug('Loading existing session for ' . $file->getId());
-			$content = null;
 			try {
 				$stateFile = $this->documentService->getStateFile($document->getId());
 				$documentState = $stateFile->getContent();
+				$this->logger->debug('Existing document, state file loaded ' . $file->getId());
 			} catch (NotFoundException $e) {
-				$this->logger->debug('State file not found for ' . $file->getId());
-				$documentState = ''; // no state saved yet.
-				// If there are no steps yet we might still need the content.
-				$steps = $this->documentService->getSteps($document->getId(), 0);
-				if (empty($steps)) {
-					$this->logger->debug('Empty steps, loading content for ' . $file->getId());
-					$content = $this->loadContent($file);
```

Comment:
This might be another cause of https://github.com/nextcloud/text/issues/5420 and is not needed as we only want to load from the file content on the initial session creation